src/main/scala/nl.knaw.dans.easy.file2bag/FilesXml.scala

```diff
@@ -31,27 +31,41 @@ object FilesXml extends DebugEnhancedLogging {
     // easy-deposit-api does not supply rights at all
     // so here we don't supply a default
     // https://github.com/DANS-KNAW/easy-deposit-api/blob/eef71618e8b776fb274da123f8011510499c741a/src/main/scala/nl.knaw.dans.easy.deposit/docs/FilesXml.scala#L40-L42
-    val itemContent = if (accessRights.isBlank) Seq[Node]()
-                      else <accessibleToRights>{ accessRights }</accessibleToRights>
-    val newItem =
-        <file filepath={ "data/" + destinationPath }>
-          <dcterms:format>{ mimeType }</dcterms:format>
-          { itemContent }
-        </file>
-    logger.info(newItem.toOneLiner)
+    var formatTagPrefix = "dcterms"
```

Comment:
@lindareijnhoudt 
We are using both `dct` and `dcterms` as prefix for the same URI through our code base. Somehow we should guarantee consistency. I suppose Vesa is trying to stick to a previously used prefix. But format is not the only potential dcterms candidate, see [fedora2vault](https://github.com/DANS-KNAW/easy-fedora2vault /blob/59fa9931a816984d4a464237d45fe53aa0280015/src/main/scala/nl/knaw/dans/easy/fedora2vault/FileItem.scala#L67-L74).

@vesaakerman 
Please avoid vars. It causes side effects and makes code harder to read. An alternative without considering the other dcterms candidates as mentioned above:

```
    val maybeFormat = (oldFilesXml \ "file" \ "format").headOption
    val defaultPrefix = "dcterms"
    val oldFilesXmlWithPossiblyAddedNamespace = maybeFormat
      .map(_ => oldFilesXml)
      .getOrElse(
        Option(oldFilesXml.scope.getURI(defaultPrefix))
          .map(uri => oldFilesXml.copy(scope = NamespaceBinding(defaultPrefix, "http://purl.org/dc/terms/", oldFilesXml.scope)))
          .getOrElse(oldFilesXml)
      )
    val formatTagPrefix = maybeFormat.map(_.prefix).getOrElse(defaultPrefix)
```

Reply:
Having looked a bit deeper into the code:

The issue asks to check for an existing namespace. I guess the cause of the problem is the inconsistent  implementation (mea culpa) of [easy-fedora2vault](https://github.com/DANS-KNAW/easy-fedora2vault/blob/59fa9931a816984d4a464237d45fe53aa0280015/src/main/scala/nl/knaw/dans/easy/fedora2vault/FileItem.scala#L28) with: `xmlns:dct="http://purl.org/dc/terms/"`
and https://github.com/DANS-KNAW/easy-add-files-to-bag/blob/f8e9a3fef6ce0c88a6704ee33c405ec6c5161265/src/main/scala/nl.knaw.dans.easy.file2bag/FilesXml.scala#L38

So a quick fix for the primary use case would be just fix that latter line.

In case `easy-add-files-to-bag` is (ab)used for bags created in another way than with `easy-fedora2vault` the check would be needed. My advise: start all over and don't look at existing format elements. They may not be there at all, for example for "thematische collectie" datasets. Check the root tag for an attribute that looks like

    xmlns:XXX="http://purl.org/dc/terms/"

The value found for XXX is the namespace prefix you should apply. If no such attribute is found, add your own.

I also see the new code adds rights. That is not mentioned in the issue description. It makes the comment about the rights obsolete so delete that comment. @lindareijnhoudt Should these rights also be added by `easy-fedora2vault`?

Reply:
see the new approach in the latest changes